packages/jazz-tools/src/subscribe/SubscriptionScope.ts

```diff
@@ -0,0 +1,523 @@
+      this.updateValue(createCoValue(this.schema, update, this));
+      this.loadChildren();
+    } else {
+      const hasChanged =
+        update.totalValidTransactions !== this.totalValidTransactions ||
+        // Checking the identity of the _raw value makes us cover the cases where the group
+        // has been updated and the coValues that don't update the totalValidTransactions value (e.g. FileStream)
+        this.value.value._raw !== update;
+
+      if (this.loadChildren()) {
+        this.updateValue(createCoValue(this.schema, update, this));
+      } else if (hasChanged) {
+        this.updateValue(createCoValue(this.schema, update, this));
+      }
+    }
+
+    this.totalValidTransactions = update.totalValidTransactions;
+
+    this.silenceUpdates = false;
+    this.triggerUpdate();
+  }
+
+  computeChildErrors() {
+    let issues: JazzErrorIssue[] = [];
+    let errorType: JazzError["type"] = "unavailable";
+
+    if (this.childErrors.size === 0 && this.validationErrors.size === 0) {
+      return undefined;
+    }
+
+    for (const value of this.childErrors.values()) {
+      errorType = value.type;
+      if (value.issues) {
+        issues.push(...value.issues);
+      }
+    }
+
+    for (const value of this.validationErrors.values()) {
+      errorType = value.type;
+      if (value.issues) {
+        issues.push(...value.issues);
+      }
+    }
+
+    return new JazzError(this.id, errorType, issues);
+  }
+
+  handleChildUpdate = (
+    id: string,
+    value: SubscriptionValue<any, any> | Unloaded,
+    key?: string,
+  ) => {
+    if (value.type === "unloaded") {
+      return;
+    }
+
+    this.childValues.set(id, value);
+
+    if (value.type === "unavailable" || value.type === "unauthorized") {
+      this.childErrors.set(id, value.prependPath(key ?? id));
+
+      this.errorFromChildren = this.computeChildErrors();
+    } else if (this.errorFromChildren && this.childErrors.has(id)) {
+      this.childErrors.delete(id);
+
+      this.errorFromChildren = this.computeChildErrors();
+    }
+
+    if (this.shouldSendUpdates()) {
+      if (this.value.type === "loaded") {
+        // On child updates, we re-create the value instance to make the updates
+        // seamless-immutable and so be compatible with React and the React compiler
+        this.updateValue(
+          createCoValue(this.schema, this.value.value._raw, this),
+        );
+      }
+    }
+
+    this.triggerUpdate();
+  };
+
+  shouldSendUpdates() {
+    if (this.value.type === "unloaded") return false;
+
+    // If the value is in error, we send the update regardless of the children statuses
+    if (this.value.type !== "loaded") return true;
+
+    for (const value of this.childValues.values()) {
+      // We don't wait for autoloaded values to be loaded, in order to stream updates
+      // on autoloaded lists or records
+      if (value.type === "unloaded" && !this.autoloaded.has(value.id)) {
+        return false;
```

Comment:
The check on `autoloaded` here is an important UX detail for the React/frameworks integration.

It makes the autoloaded deps "stream" results as opposed as waiting for all the autoloaded deps before triggering a single update.

We could make the behavoir configurable to support the cases where it's better to reduce the number of updates (e.g. when subscribing in a worker)

Reply:
I think this is basically equivalent to the current autoload behaviour - and you're exactly correct about the tradeoff: stuff appears quicker but there is a danger of overloading the framework with updates to render.

Let's see what makes sense once we remove autoload. Probably this interacts with pagination as well where you typically want to wait for a whole page of items, but not more.